src/file_pipe_log/pipe_builder.rs

```diff
@@ -528,6 +567,25 @@ impl<F: FileSystem> DualPipesBuilder<F> {
         )?;
         DualPipes::open(self.dir_locks, appender, rewriter)
     }
+
+    fn open(&self, file_name: &FileName, is_last_one: bool) -> Result<File<F>> {
+        let perm = if file_name.recycled
```

Comment:
No need to write recycled? recycled file will be reopened with the right permission before write.